W-10739673-batchJob-duke (#2182)
mulesoft/docs-mule-runtime: modules/ROOT/pages/batch-processing-concept.adoc

image::batch-diagram.jpg[batch+diagram]

> Note that by default, a batch job instance does not wait for all its queued records to finish processing in one batch step before pushing any of them to the next batch step. Configuring an aggregator can change that behavior:

Reviewer: "can change" or "changes" ? I think the latter is more accurate.

Author: does it always change the behavior?

> Note that by default, a batch job instance does not wait for all its queued records to finish processing in one batch step before pushing any of them to the next batch step. However,  configuring the batch job or the aggregator can change the behavior of the batch job step:

Reviewer 2: ```suggestion
Note that by default, a batch job instance does not wait for all its queued records to finish processing in one batch step before pushing any of them to the next batch step. Configuring an aggregator can change that behavior:
```

> Note that by default, a batch job instance does not wait for all its queued records to finish processing in one batch step before pushing any of them to the next batch step. However, you can configure the batch job or the block size of the aggregator to change the behavior of the batch job step:

Reviewer 2: I think "you can configure the batch job or the block size of the aggregator" is not correct because you are always configuring an aggregator and not only the batch job. Maybe just a rephrase of the original text will be better?

Author: revised

> Note that by default, a batch job instance does not wait for all its queued records to finish processing in one batch step before pushing any of them to the next batch step. However, you can configure the batch job or the block size of the aggregator to change the behavior of the batch job step:

Author: ```suggestion
Note that by default, a batch job instance does not wait for all its queued records to finish processing in one batch step before pushing any of them to the next batch step. However,  configuring the batch job or the aggregator can change the behavior of the batch job step:
```

* Behavior with aggregator configured with fixed size
+
In this scenario, the batch step sends the processed records to an aggregator, which starts processing the records and buffering them until the configured aggregator's size is reached. After that, the aggregator sends the aggregated records to the stepping queue.
+
image:mruntime-batch-job-process-aggregator-fixed.png[Batch job process with Aggregator with fixed size]
+

> <1> The batch job builds blocks of records limited by the configured block size and sends them to their corresponding batch step for processing. If fewer records are available than the configured size, the batch job will also send them to the corresponding batch step instead of waiting for more records.

Reviewer: "will also send" > "sends"

Author: ```suggestion
<1> The batch job builds blocks of records limited by the configured block size and sends them to their corresponding batch step for processing. If fewer records are available than the configured size, the batch job sends them to the corresponding batch step instead of waiting for more records.
```

> <1> The batch job builds blocks of records limited by the configured block size and sends them to their corresponding batch step for processing. The block size is an upper limit. If fewer records are available than the configured size, the batch job processes the records. 

Author: ```suggestion
<1> The batch job builds blocks of records limited by the configured block size and sends them to their corresponding batch step for processing. The block size is an upper limit. If fewer records are available than the configured size, the batch job also sends the records to the batch step. 
```

> <1> The batch job builds blocks of records limited by the configured block size and sends them to their corresponding batch step for processing. The block size is an upper limit. If fewer records are available than the configured size, the batch job also sends the records to the batch step. 

Reviewer 2: ```suggestion
<1> The batch job builds blocks of records limited by the configured block size and sends them to their corresponding batch step for processing. If fewer records are available than the configured size, the batch job will also send them to the corresponding batch step instead of waiting for more records.
```

> <1> The batch job builds blocks of records limited by the configured block size and sends them to their corresponding batch step for processing. The block size is an upper limit. If fewer records are available than the configured size, the batch job processes the records. 

Reviewer 2: Maybe "the batch job processes the records" can be "the batch job will also send them to the corresponding batch step instead of waiting for more records".
